src/components/interface/capmgr/memmgr.h

```diff
@@ -10,4 +11,23 @@ int memmgr_shared_page_alloc(vaddr_t *pgaddr);
 int memmgr_shared_page_allocn(int num_pages, vaddr_t *pgaddr);
 int memmgr_shared_page_map(int id, vaddr_t *pgaddr);
 
+/* This magic number is double the tls size defined in RK */
+#define TLS_AREA_SIZE 32
+#define TLS_NUM_PAGES (round_up_to_page(TLS_AREA_SIZE * MAX_NUM_THREADS) / PAGE_SIZE)
+#define TLS_BASE_ADDR 0x70000000
+
+void *memmgr_tls_alloc(unsigned int dst_tid);
+void *_memmgr_tls_alloc_and_set(void *area);
+
+static void *
+memmgr_tls_alloc_and_set(void *area)
```

Comment:
As far as i can tell area is never used

Reply:
It is only used above. As @phanikishoreg said, we could leave it up to the interface user to set what addr points to on their own.